### PR TITLE
fix(odd): pretend to be online all the time

### PR DESCRIPTION
**Pull request by sfoster1**
For some reason, when the ODD starts and the robot boots while not connected to any network, some (some!) requests to localhost will fail. It's really strange. One thing that seems to help is pretending to be online by overriding the `navigator.onLine` browser api to always return true.

## Testing
- Boot the odd with no network connected and see if stuff works I guess

Closes RQA-1754

### CODE REVIEW

File: app-shell-odd/Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ setup:
 
 .PHONY: clean
 clean:
-	shx rm -rf lib dist
+	shx rm -rf lib dist opentrons-robot-app.tar.gz
```
Comment (shlokamin) [inline app-shell-odd/Makefile:51]: lol ty

File: app/src/App/OnDeviceDisplayApp.tsx [hunk 2]
```diff
@@ -47,9 +47,13 @@ import { useCurrentRunRoute, useProtocolReceiptToast } from './hooks'
 
 import { OnDeviceDisplayAppFallback } from './OnDeviceDisplayAppFallback'
 
+import { hackWindowNavigatorOnLine } from './hacks'
+
 import type { Dispatch } from '../redux/types'
 import type { RouteProps } from './types'
 
+hackWindowNavigatorOnLine()
+
```
Comment (shlokamin) [inline app/src/App/OnDeviceDisplayApp.tsx:58]: can we throw a comment before this fn call?

File: app/src/App/OnDeviceDisplayApp.tsx [hunk 3]
```diff
@@ -257,7 +261,7 @@ export const OnDeviceDisplayApp = (): JSX.Element => {
 
   // TODO (sb:6/12/23) Create a notification manager to set up preference and order of takeover modals
   return (
-    <ApiHostProvider hostname="localhost">
+    <ApiHostProvider hostname="127.0.0.1">
```
Comment (shlokamin) [inline app/src/App/OnDeviceDisplayApp.tsx:266]: i guess this doesnt hurt but probably not necessary anymore

**Review by shlokamin — APPROVED** [review]
lgtm, what a journey

File: app/src/App/OnDeviceDisplayApp.tsx (hunk 3 above)
Comment (sfoster1) [reply app/src/App/OnDeviceDisplayApp.tsx:266]: yeah... i like having it though

**Pull request merged**